src/main/java/com/mineinabyss/mobzy/spawning/SpawnTask.kt

```diff
@@ -26,14 +26,14 @@ class SpawnTask : BukkitRunnable() {
 
             val container = WorldGuard.getInstance().platform.regionContainer
 
-            Bukkit.getOnlinePlayers().toList().toPlayerGroups().forEach { playerGroup->
+            Bukkit.getOnlinePlayers().toList().toPlayerGroups().forEach { playerGroup ->
                 val regionManager = container[BukkitAdapter.adapt(playerGroup[0].world)] ?: return@forEach
                 val toSpawn: MutableList<MobSpawnEvent> = mutableListOf()
 
                 //STEP 1: Generate array of ChunkSpawns around player group
-               val spawnChunkGrid = SpawnChunkGrid(playerGroup.map { it.location }, MobzyConfig.minChunkSpawnRad, MobzyConfig.maxChunkSpawnRad)
+                val spawnChunkGrid = SpawnChunkGrid(playerGroup.map { it.location }, MobzyConfig.minChunkSpawnRad, MobzyConfig.maxChunkSpawnRad)
 
-                val customMobs = spawnChunkGrid.allChunks.customMobs
+                val customMobs = spawnChunkGrid.allChunks.filter { it.isLoaded }.customMobs
```

Comment:
makes sense. are you sure it's needed?

Reply:
Timings report showed some lag spikes coming from mobzy related to chunks. I think loading up the unloaded ones synchronously was causing them. Guess we'll see how much it helps soon.